File: Level(2)open/Level(2)open.js
function solution(record) {
    const splitRecord = [];
    const uidNickname = {};
    var answer = [];
    record.forEach(r => {
        const s = r.split(' ');
        splitRecord.push([s[0], s[1]])
        if(s[2]) {
            uidNickname[s[1]] = s[2];
        }
    });
    splitRecord.forEach(r => {
        let msg;
        if(r[0] === 'Enter') {
            msg = '님이 들어왔습니다.'
        }
        else if(r[0] === 'Leave') {
            msg =  '님이 나갔습니다.'
        }
        if(msg) {
            answer.push(`${uidNickname[r[1]]}${msg}`);
        }
    })

    return answer;
}

File: Level(2)open/Level(2)open.py
def solution(record):
    ENTER_SUFFIX = "님이 들어왔습니다." 
    LEAVE_SUFFIX = "님이 나갔습니다."

    uidToNickname = {}
    behavior_list = []

    for r in record: 
        action = r.split(' ')[0]
        uid = r.split(' ')[1]

        if action == "Enter":
            uidToNickname[uid] = r.split(' ')[2]
            behavior_list.append((uid, "e"))
        elif action == "Change":
            uidToNickname[uid] = r.split(' ')[2]
        elif action == "Leave":
            behavior_list.append((uid, "l"))

    answer = []
    for b in behavior_list:
        if b[1] == "e":
            answer.append(uidToNickname[b[0]] + ENTER_SUFFIX)
        elif b[1] == "l":
            answer.append(uidToNickname[b[0]] + LEAVE_SUFFIX)

    return answer
